skcosmo/utils/_pcovr_utils.py

```diff
@@ -1,7 +1,62 @@
+from copy import deepcopy
+
 import numpy as np
-from scipy import linalg
+from sklearn.base import clone
+from sklearn.exceptions import NotFittedError
 from sklearn.metrics.pairwise import pairwise_kernels
 from sklearn.utils.extmath import randomized_svd
+from sklearn.utils.validation import check_is_fitted
+
+
+def check_lr_fit(regressor, X, y=None):
+    r"""
+    Checks that an regressor is fitted, and if not,
+    fits it with the provided data
+
+    :param regressor: sklearn-style regressor
+    :type regressor: object
+    :param X: feature matrix with which to fit the regressor
+        if it is not already fitted
+    :type X: array
+    :param y: target values with which to fit the regressor
+        if it is not already fitted
+    :type y: array
+    :param sample_weight: sample weights with which to fit
+        the regressor if not already fitted
+    :type sample_weight: array of shape (n_samples,)
+    """
+    try:
+        check_is_fitted(regressor)
+        fitted_regressor = deepcopy(regressor)
```

Comment:
do we need to copy the regressor?

Reply:
I suppose we don't _have_ to, but I think copying is the safe choice. We don't make any modifications to the fitted regressor within PCovR, but if we did, I'm guessing it would propagate to the original object, outside the `PCovR` instance. I see the copying as an insurance policy to make sure that we avoid this and make the class more robust to future tinkering.

Reply:
Bof, okay, I can agree to that.